bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/Canopus.java

```diff
@@ -227,154 +152,99 @@ public GuideStarValidation validate(SPTarget guideStar, ObsContext ctx) {
-                    .map(base -> {
-                        CoordinateDiff diff = new CoordinateDiff(base, gscoords);
-                        // Get offset and switch it to be defined in the same coordinate
-                        // system as the shape.
-                        Offset dis = diff.getOffset();
-                        double p = -dis.p().toArcsecs().getMagnitude();
-                        double q = -dis.q().toArcsecs().getMagnitude();
-                        Set<Offset> offsets = new TreeSet<>();
-                        offsets.add(offset);
-
-                        Area a = Canopus.instance.offsetIntersection(ctx, offsets);
-                        return a != null && a.contains(p, q);
-                    }))
-                .getOrElse(false);
+        public boolean inRange(final ObsContext ctx, final Offset offset) {
+            return ctx.getTargets().getPrimaryGuideProbeTargets(this)
+                    .flatMap(GuideProbeTargets::getPrimary)
+                    .flatMap(gs -> gs
+                            .getSkycalcCoordinates(ctx.getSchedulingBlockStart())
+                            .flatMap(gscoords -> ctx.getBaseCoordinates()
+                                    .map(base -> {
+                                        final CoordinateDiff diff = new CoordinateDiff(base, gscoords);
+
+                                        // Get offset and switch it to be defined in the same coordinate
+                                        // system as the shape.
+                                        final Offset dis = diff.getOffset();
+                                        final double p = -dis.p().toArcsecs().getMagnitude();
+                                        final double q = -dis.q().toArcsecs().getMagnitude();
+                                        final Set<Offset> offsets = new TreeSet<>();
+                                        offsets.add(offset);
+
+                                        final Area a = Canopus.instance.offsetIntersection(ctx, offsets);
+                                        return a != null && a.contains(p, q);
+                                    })))
+                    .getOrElse(false);
         }
     }
 
-        //     REL-1042: Update the Canopus probe limits in the OT
-//    public static final double RADIUS_ARCSEC = 60.0;
     public static final double RADIUS_ARCSEC = 62.5;
 
-
-    // Probe ranges expressed in arcsec
-//    private static final Tuple2<Integer, Integer> PROBE1_RANGE = new Pair<Integer, Integer>(-66, 24);
-//    private static final Tuple2<Integer, Integer> PROBE2_RANGE = new Pair<Integer, Integer>(-24, 66);
-
-    // REL-297:
     // P1 can access a square field of 3.7 x 3.0 armcin, but the center is at (+0.11,+0.3)
     // P2 can access a square field of 4.2 x 3.0 arcmin, but the center is at (0.0,-0.33)
     private static final Point2D.Double PROBE1_CENTER = new Point2D.Double(0.11 * 60.0, 0.3 * 60.0);
```

Comment:
We don't need these definitions any more either?

Reply:
They were commented out in the original code, so I just removed them since they were no longer being used.

Reply:
Ooops. I understand now what you meant: we don't need these other definitions anymore.